crates/driver/src/domain/competition/solution/settlement.rs

```diff
@@ -249,7 +249,7 @@ impl Settlement {
                     mempools::RevertProtection::Disabled => gas_cost,
                 };
                 let score = competition::Score::new(
-                    auction.score_cap(),
+                    competition::Score(eth::U256::MAX.try_into().unwrap()),
```

Comment:
Maybe it makes sense to remove this `Score` think in this PR as well?

Reply:
The code change becomes too large then and hard to review. Planned to do it as part of https://github.com/cowprotocol/services/issues/2493